app/agents_v2/primary_agent/troubleshooting/diagnostic_sequencer.py

```
@@ -0,0 +1,512 @@
+        if session.customer_emotional_state in [EmotionalState.CONFUSED, EmotionalState.ANXIOUS]:
+            return False
+        
+        # Increase after multiple successful steps
+        if len(session.completed_steps) >= 3:
+            return True
+        
+        # Increase for technical customers
+        if session.customer_technical_level >= 4 and len(session.completed_steps) >= 2:
+            return True
+        
+        return False
```

Ruff (0.11.9)

Return the condition directly

Inline condition

(SIM103)